### DISCUSSION TITLE
What does “indexSurface” mean about "SatelliteModelCreate" modules

### DISCUSSION THREAD

**Discussion by LNTU-YS** [groops-devs/groops#263] [General]
Hello,

I am trying to create a satellite model, but I don't understand what “SatelliteModelCreate——satellite——module——solarPanel——indexSurface” means. I noticed that the annotation for “indexSurface”is “index of solar panel surfaces”, I referred to some papers but couldn't find it.

![image](https://github.com/groops-devs/groops/assets/111558200/170bba0f-e316-4d6b-9121-b8519c9e712a)

And according to my understanding, I have already set up the solar panel once in the "surfaces" module, why do I need to set it up again in the "module" module? This makes me unsure if my understanding is correct.

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Comment by LNTU-YS** [comment]
I seem to have solved this problem. It should be the number of rows representing the solar panel information in the “surfaces”module. But I have new questions about “rotationAxis” and “normal”. I noticed that in Sentinel's satellite model, both of these parts have values such as -1, 0, and 1. So is it related to body-+/-XYZ?

**Comment by tmayerguerr** [comment]
Hi YUAN,
You are right, the indices refers to the surfaces from above (starting from zero). The indexed surfaces of the solar panel  are rotated around the rotationAxis towards the sun (the provided normal vector points then as close as possible towards the sun). All coordinates must be given in the satellite reference system.
regards
Torsten
